mindsdb_sdk/agents.py

```diff
@@ -0,0 +1,188 @@
+from requests.exceptions import HTTPError
+from typing import List, Union
+import datetime
+
+from mindsdb_sdk.models import Model
+from mindsdb_sdk.skills import Skill, Skills
+from mindsdb_sdk.utils.objects_collection import CollectionBase
+
+
+class AgentCompletion:
+    """Represents a full MindsDB agent completion"""
+    def __init__(self, content: str):
+        self.content = content
+
+    def __repr__(self):
+        return self.content
+
+
+class Agent:
+    """Represents a MindsDB agent"""
+    def __init__(
+            self,
+            name: str,
+            model_name: str,
+            skills: List[Skill],
+            params: dict,
+            created_at: datetime.datetime,
+            updated_at: datetime.datetime,
+            collection: CollectionBase = None
+            ):
+        self.name = name
+        self.model_name = model_name
+        self.skills = skills
+        self.params = params
+        self.created_at = created_at
+        self.updated_at = updated_at
+        self.collection = collection
+
+    def completion(self, messages: List[dict]) -> AgentCompletion:
+        return self.collection.completion(self.name, messages)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(name: {self.name})'
+
+    def __eq__(self, other):
+        if self.name != other.name:
```

Comment:
how this method should be used. in which use-case user compares this objects? 
if it is required: from mindsdb server perspective two agents can't have the same name in one project. So name-project_name is already unique identifier.

Reply:
The main idea: skill and agent can be identified by name (and project), we can't have the same name in one project. So we don't need to compare all columns to check if it is the same object in mindsdb. 
If the purpose is only for tests (to compare result with expectation) we could move this comparator to tests (also we can do it later)